257 user docs (#259)
Consensys/linea-constraints: README.md

# Linea constraints 

This repository hosts the implementation of the constraint system from the Linea zk-EVM specification.

It serves developers by making the Linea tech stack open source under 
the [Apache 2.0 license](LICENSE).

## What is Linea?

[Linea](https://linea.build) is a developer-ready layer 2 network scaling Ethereum. It's secured with a zero-knowledge rollup, built on lattice-based cryptography, and powered by [Consensys](https://consensys.io).

<!-- ## Get started

tbd -->

## Looking for the Linea code?

Linea's stack is made up of multiple repositories, these include:

> 

Author: ```suggestion

- [linea-constraints](https://github.com/Consensys/linea-constraints): 
> Implementation of the constraint system specified in the [linea-specification repo](https://github.com/Consensys/linea-specification).
>
> Linea’s constraint system applies to so-called traces which are large matrices of fixed width (i.e. fixed number of columns or ‘registries’) and variable depth (correlating with the complexity of the EVM execution). The production of such traces is the job of the [linea-tracer repo](https://github.com/Consensys/linea-tracer).
>
> Constraints and traces are two of the inputs to the prover.
```

# Linea constraints 

> This repository hosts the implementation of the constraint system from the Linea zk-EVM specification.

Author: ```suggestion
This repository hosts the Implementation of the constraint system specified in the [linea-specification repo](https://github.com/Consensys/linea-specification). 

Linea’s constraint system applies to so-called traces which are large matrices of fixed width (i.e. fixed number of columns or ‘registries’) and variable depth (correlating with the complexity of the EVM execution). The production of such traces is the job of the [linea-tracer repo](https://github.com/Consensys/linea-tracer).

Constraints and traces are two of the inputs to the prover.
```

It serves developers by making the Linea tech stack open source under 
the [Apache 2.0 license](LICENSE).

## What is Linea?

[Linea](https://linea.build) is a developer-ready layer 2 network scaling Ethereum. It's secured with a zero-knowledge rollup, built on lattice-based cryptography, and powered by [Consensys](https://consensys.io).

<!-- ## Get started

tbd -->

## Looking for the Linea code?

Linea's stack is made up of multiple repositories, these include:

- [linea-monorepo](https://github.com/Consensys/linea-monorepo): The main repository for the Linea stack & network 
- [linea-besu](https://github.com/Consensys/linea-besu): Fork of Besu to implement the Linea-Besu client
- [linea-sequencer](https://github.com/Consensys/linea-sequencer): A set of Linea-Besu plugins for the sequencer and RPC nodes
- [linea-tracer](https://github.com/Consensys/linea-tracer): Linea-Besu plugin which produces the traces that the constraint system applies and that serve as inputs to the prover

> - [linea-constraints](https://github.com/Consensys/linea-constraints): Implementation of the constraint system from the specification

Author: ```suggestion
```

# Linea constraints 

This repository hosts the implementation of the constraint system from the Linea zk-EVM specification.

It serves developers by making the Linea tech stack open source under 
the [Apache 2.0 license](LICENSE).

## What is Linea?

[Linea](https://linea.build) is a developer-ready layer 2 network scaling Ethereum. It's secured with a zero-knowledge rollup, built on lattice-based cryptography, and powered by [Consensys](https://consensys.io).

<!-- ## Get started

tbd -->

## Looking for the Linea code?

Linea's stack is made up of multiple repositories, these include:

- [linea-constraints](https://github.com/Consensys/linea-constraints): 

> > Constraints and traces are two of the inputs to the prover.

Author: ```suggestion
```

# Linea constraints 

This repository hosts the implementation of the constraint system from the Linea zk-EVM specification.

It serves developers by making the Linea tech stack open source under 
the [Apache 2.0 license](LICENSE).

## What is Linea?

[Linea](https://linea.build) is a developer-ready layer 2 network scaling Ethereum. It's secured with a zero-knowledge rollup, built on lattice-based cryptography, and powered by [Consensys](https://consensys.io).

<!-- ## Get started

tbd -->

## Looking for the Linea code?

Linea's stack is made up of multiple repositories, these include:

- [linea-constraints](https://github.com/Consensys/linea-constraints): 

> >

Author: ```suggestion
```

# Linea constraints 

This repository hosts the implementation of the constraint system from the Linea zk-EVM specification.

It serves developers by making the Linea tech stack open source under 
the [Apache 2.0 license](LICENSE).

## What is Linea?

[Linea](https://linea.build) is a developer-ready layer 2 network scaling Ethereum. It's secured with a zero-knowledge rollup, built on lattice-based cryptography, and powered by [Consensys](https://consensys.io).

<!-- ## Get started

tbd -->

## Looking for the Linea code?

Linea's stack is made up of multiple repositories, these include:

- [linea-constraints](https://github.com/Consensys/linea-constraints): 

> > Linea’s constraint system applies to so-called traces which are large matrices of fixed width (i.e. fixed number of columns or ‘registries’) and variable depth (correlating with the complexity of the EVM execution). The production of such traces is the job of the [linea-tracer repo](https://github.com/Consensys/linea-tracer).

Author: ```suggestion
```

# Linea constraints 

> This repository hosts the Implementation of the constraint system specified in the [linea-specification repo](https://github.com/Consensys/linea-specification). 

Author: ```suggestion
This repository hosts the implementation of the constraint system specified in the [linea-specification repo](https://github.com/Consensys/linea-specification). 
```

It serves developers by making the Linea tech stack open source under 
the [Apache 2.0 license](LICENSE).

## What is Linea?

[Linea](https://linea.build) is a developer-ready layer 2 network scaling Ethereum. It's secured with a zero-knowledge rollup, built on lattice-based cryptography, and powered by [Consensys](https://consensys.io).

<!-- ## Get started

tbd -->

## Looking for the Linea code?

Linea's stack is made up of multiple repositories, these include:

- [linea-constraints](https://github.com/Consensys/linea-constraints): 

> >

Author: ```suggestion
```

Linea's stack is made up of multiple repositories, these include:

> - [linea-constraints](https://github.com/Consensys/linea-constraints): 

Author: ```suggestion
- This repo, [linea-constraints](https://github.com/Consensys/linea-constraints): Implementation of the constraint system from the specification
```

# Linea constraints 

This repository hosts the implementation of the constraint system from the Linea zk-EVM specification.

It serves developers by making the Linea tech stack open source under 
the [Apache 2.0 license](LICENSE).

## What is Linea?

[Linea](https://linea.build) is a developer-ready layer 2 network scaling Ethereum. It's secured with a zero-knowledge rollup, built on lattice-based cryptography, and powered by [Consensys](https://consensys.io).

<!-- ## Get started

tbd -->

## Looking for the Linea code?

Linea's stack is made up of multiple repositories, these include:

- [linea-constraints](https://github.com/Consensys/linea-constraints): 

> > Implementation of the constraint system specified in the [linea-specification repo](https://github.com/Consensys/linea-specification).

Author: ```suggestion
```
